labs/b9.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Lab 9 - Containers and Configuration Management
+layout: lab
+---
+
+## Overview
+
+This lab is designed to give you some hands-on experience with Docker and Puppet! By the end of this assignment, you should be able to:
+
+- Create and use a Docker container interactively  and create a Dockerfile, which allows you to declaratively define your containers.
+- Write a basic Puppet manifest and apply the configuration to your VM.
+
+To submit the lab, answer the questions in [this Google form](https://docs.google.com/forms/d/e/1FAIpQLSfZcfT3tDtQV8ecEjgvLDcb6uEnV4Q57BAw949InCkvjG5lYA/viewform?usp=sf_link)
+
+#### Installing Docker
+
+One of the annoying things about Docker is it’s usually not in `apt` and, if it is, it’s always under a different name. Here’s a link to the .deb for Docker Community Edition you can download and install on your VM: [https://download.docker.com/linux/debian/dists/stretch/pool/stable/amd64/docker-ce\_18.03.0~ce-0~debian\_amd64.deb](https://download.docker.com/linux/debian/dists/stretch/pool/stable/amd64/docker-ce_18.03.0~ce-0~debian_amd64.deb).
+
+I trust by this point in the course installing this on your Linux machine should be a breeze :).
+
+Hint: use `wget` and `dpkg`.
+
+After installing, I recommend running `sudo usermod -aG docker $USER`. This adds your user to the docker group so you can run `docker` as a non-root user. This means you won’t have to type `sudo docker` all the time. This is optional but for the rest of the lab I’m going to assume that you did this. You might see output like:
```

Comment:
I'd say here they have to relog to get it to apply, but otherwise this should be good.